packages/option/src/index.ts

```diff
@@ -0,0 +1,181 @@
+import { createSignal } from "solid-js";
+
+/**
+ * a utility class representing an Optional value, which can be either Some\<T> or None.
+ *
+ * @example
+ * ```ts
+ * const option = Some(5);
+ *
+ * if (option.isSome()) {
+ *   doSomething(option.unwrap())
+ * }
+ * ```
+ */
+class Option<T> {
+  #value: T | null;
+  constructor(value: T | null) {
+    this.#value = value;
+  }
+  /**
+   * Checks whether the option is the None variant.
+   * @example
+   * ```ts
+   * const userInput = None();
+   *
+   * if (userInput.isNone()) return alert("Invalid input");
+   * ```
+   */
+  isNone() {
+    return this.#value == null;
+  }
+  /**
+   * Checks whether the option is the Some variant.
+   * @example
+   * ```ts
+   * const userSubmission = Some(5);
+   *
+   * if (userSubmission.isSome()) return alert(`The user has already submitted value of ${userInput}`);
+   * ```
+   */
+  isSome() {
```

Comment:
Not sure if I get the point of having these methods attached to the object.
Wouldn't it be better to export just a normal function for checking that, it would also offer better ts narrowing support.
```ts
function isSome<T>(v: T | null | undefined): v is T {
  return v != null
}

declare x: number | null

if (isSome(x)) {
  x // T: number
}
```

Functions also are better tree-shakeable.